exercises/simple-linked-list/src/test/scala/SimpleLinkedListTest.scala

```diff
@@ -0,0 +1,56 @@
+import org.scalatest.{FlatSpec, Matchers}
+
+class SimpleLinkedListTest extends FlatSpec with Matchers {
+
+  it should "handle single item list" in {
+    val list = SimpleLinkedList().add(1)
+    list.isEmpty should be (false)
+    list.value should be (1)
+  }
+
+  it should "handle single item has no next item" in {
```

Comment:
We would need a newly defined format for the canonical data to state those "properties" or "laws" and have them automatically generated.
But I think it could be worth the effort as there are a couple of languages with property-based testing libraries.

I have made up some properties for this exercise in the Haskell track, see [here](https://github.com/abo64/xhaskell/blob/cfc41756a0b4e9b742e08d27c4c107434f549934/exercises/simple-linked-list/test/Tests.hs#L91). We could take those and translate them to ScalaTest/ScalaCheck.